### PR TITLE
guard kubeadm dependencies on k8s.io/kubernetes

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
kubeadm trimmed all their dependencies on k8s.io/kubernetes! this ensures we don't accidentally pick up new ones

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area kubeadm
/cc @neolit123

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks, @liggitt 
looks like we forgot to clean import boss, @rosti 

/lgtm

**Review by yastij — COMMENTED** [review]
/lgtm

**Comment by neolit123** [comment]
i think it needs bazel update?

```
diff -Naupr -x _output /home/prow/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient/BUILD /tmp/verify-bazel.HWSOcb/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient/BUILD
--- /home/prow/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient/BUILD	2019-11-13 19:03:38.630498954 +0000
+++ /tmp/verify-bazel.HWSOcb/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient/BUILD	2019-11-13 19:04:06.194932288 +0000
@@ -19,7 +19,6 @@ go_library(
     deps = [
         "//cmd/kubeadm/app/constants:go_default_library",
         "//cmd/kubeadm/app/util:go_default_library",
-        "//pkg/kubelet/types:go_default_library",
         "//staging/src/k8s.io/api/apps/v1:go_default_library",
         "//staging/src/k8s.io/api/core/v1:go_default_library",
         "//staging/src/k8s.io/api/rbac/v1:go_default_library",

Run ./hack/update-bazel.sh
```

File: cmd/kubeadm/app/util/apiclient/wait.go [hunk 1]
```diff
@@ -24,14 +24,13 @@ import (
...
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	netutil "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
-	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
```
Comment (yastij) [inline cmd/kubeadm/app/util/apiclient/wait.go]: @neolit123 - yup needs update bazel, this was removed

Comment (liggitt) [reply cmd/kubeadm/app/util/apiclient/wait.go]: bah. updated.

**Comment by liggitt** [comment]
updated bazel, retagged

File: cmd/kubeadm/app/util/apiclient/wait.go [hunk 2]
```diff
@@ -246,7 +245,7 @@ func getStaticPodSingleHash(client clientset.Interface, nodeName string, compone
 		return "", err
 	}
 
-	staticPodHash := staticPod.Annotations[kubetypes.ConfigHashAnnotationKey]
+	staticPodHash := staticPod.Annotations["kubernetes.io/config.hash"]
```
Comment (rosti) [inline cmd/kubeadm/app/util/apiclient/wait.go]: Thanks @liggitt !
Will make this a const once master is reopened for the 1.18 cycle.
/lgtm

**Pull request merged**